File: cs/ch04/Money/Dollar.cs
﻿namespace TheSoftwareGorilla.TDD.Money;

public class Dollar
{
    public int Amount { get; }

    public Dollar(int amount)
    {
        Amount = amount;
    }

    public Dollar Times(int multiplier)
    {
        return new Dollar(Amount * multiplier);
    }

    public override bool Equals(object? obj)
    {
        return obj is Dollar dollar && Amount == dollar.Amount;
    }
}

File: java/ch03/src/main/java/com/thesoftwaregorilla/tdd/money/Dollar.java
package com.thesoftwaregorilla.tdd.money;

public class Dollar {
    int amount;

    Dollar(int amount) {
        this.amount = amount;
    }

    public Dollar times(int multiplier) {
        return new Dollar(amount * multiplier);
    }

    public boolean equals(Object object) {
        return object instanceof Dollar dollar && this.amount == dollar.amount;
    }
}
